### PR TITLE
fix Oracle query limit compliance

### PR DESCRIPTION
**Pull request by to-kn**
these changes fixed the Problems with #1702 for me

### CODE REVIEW

**Review by fschrempf — CHANGES_REQUESTED** [review]
Thanks for the contribution! I think it would be better to add the chunking directly in `getFoldersForGroups()` instead of implementing it in multiple places. And a comment with the reason for the chunking would be nice.

**Comment by fschrempf** [comment]
You can have a look at https://github.com/nextcloud/server/pull/27187 or https://github.com/nextcloud/mail/pull/4947 for other places where this was fixed. I also opened an issue (#1805) to check any other queries that might hit the limitations.

**Comment by to-kn** [comment]
HI @fschrempf do you mean like that? To be honest i did not had the possibility to validate the fix like that...

**Review by fschrempf — CHANGES_REQUESTED** [review]
The code looks good to me. If you could add the comment I was asking for and squash the two commits into a single one, I think we are good to go.

**Comment by fschrempf** [comment]
And we nee to find out why the unit tests fail. Might not be related to this PR, I haven't checked. Maybe rebasing already does the trick.

**Comment by to-kn** [comment]
@fschrempf should be good

File: lib/Folder/FolderManager.php [hunk 1]
```diff
@@ -631,7 +637,7 @@ public function setFolderACL(int $folderId, bool $acl): void {
 	 */
 	public function getFoldersForUser(IUser $user, int $rootStorageId = 0): array {
 		$groups = $this->groupManager->getUserGroupIds($user);
-		$folders = $this->getFoldersForGroups($groups, $rootStorageId);
+        $folders = $this->getFoldersForGroups($groups, $rootStorageId);
```
Comment (CarlSchwan) [inline lib/Folder/FolderManager.php:640]:
```suggestion
       $folders = $this->getFoldersForGroups($groups, $rootStorageId);
```

File: lib/Folder/FolderManager.php [hunk 2]
```diff
@@ -654,7 +660,7 @@ public function getFoldersForUser(IUser $user, int $rootStorageId = 0): array {
 	 */
 	public function getFolderPermissionsForUser(IUser $user, int $folderId): int {
 		$groups = $this->groupManager->getUserGroupIds($user);
-		$folders = $this->getFoldersForGroups($groups);
+        $folders = $this->getFoldersForGroups($groups);
```
Comment (CarlSchwan) [inline lib/Folder/FolderManager.php:663]:
```suggestion
       $folders = $this->getFoldersForGroups($groups);
```

**Review by CarlSchwan — APPROVED** [review]
Code looks fine, let's wait for the tests results

**Review by CarlSchwan — CHANGES_REQUESTED** [review]
Tests are still failing and it seems to be directly related to this PR

File: lib/Folder/FolderManager.php [hunk 3]
```diff
@@ -453,10 +453,16 @@ public function getFoldersForGroups(array $groupIds, int $rootStorageId = 0): ar
...
 			)
-			->where($query->expr()->in('a.group_id', $query->createNamedParameter($groupIds, IQueryBuilder::PARAM_STR_ARRAY)));
+			->where($query->expr()->in('a.group_id', $query->createParameter('groupIds')));
 		$this->joinQueryWithFileCache($query, $rootStorageId);
 
-		$result = $query->executeQuery()->fetchAll();
+        // add chunking because Oracle can't deal with more than 1000 values in an expression list for in queries.
+        $result = [];
+        foreach (array_chunk($groupIds, 1000) as $chunk) {
+            $query->setParameter('groupIds', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
```
Comment (CarlSchwan) [inline lib/Folder/FolderManager.php:462]:
```suggestion
            $query->setParameter('groupIds', $chunk, IQueryBuilder::PARAM_STR_ARRAY);
```

Probably should fix the tests

**Comment by to-kn** [comment]
@CarlSchwan thank you for your feed-back. I did not manage to get the Tests running locally so i was not able to check it beforehand...

**Comment by to-kn** [comment]
@CarlSchwan since tests are running now, i sqashed the commits again. Let me know if anything is missing

**Review by CarlSchwan — APPROVED** [review]
LGTM

**Review by fschrempf — APPROVED** [review]

**Pull request merged**